pytorch_lightning/trainer/trainer.py

```diff
@@ -665,6 +665,8 @@ def run_evaluation(self, on_epoch=False):
         # ref model
         model = self.lightning_module
         model.zero_grad()
+
+        self.model.eval()
```

Comment:
careful, right above here we have a special hook `self.evaluation_loop.on_evaluation_model_eval`
we probably don't want to override this behaviour if the user implements this hook
same for the other places (e.g. predict)
confidence on this statement: ~60%